x-pack/plugins/observability/public/components/threshold/components/alert_details_app_section.tsx

```diff
@@ -208,24 +199,85 @@ export default function AlertDetailsAppSection({
     </>
   ) : null;
 
+  const cpuMetricPrefix = 'system.cpu';
+  const memoryMetricPrefix = 'system.memory';
+  const relatedMetrics = [
+    'system.cpu.user.pct',
+    'system.load.1',
+    'system.memory.actual.used.pct',
+    'system.filesystem.used.pct',
+    'host.network.ingress.bytes',
+    'host.network.egress.bytes',
+  ];
+  const fnList = ['avg', 'sum', 'min', 'max'];
+
+  const isCpuOrMemoryCriterion = (criterion: MetricExpression) =>
+    criterion.metrics?.some(
+      (metric: CustomThresholdExpressionMetric) =>
+        metric.field?.includes(cpuMetricPrefix) || metric.field?.includes(memoryMetricPrefix)
+    );
+
+  const relatedMetricsPerCriteria = () => {
```

Comment:
Can we have a test for this function? We have a lot of checks like `isCpuOrMemoryCriterion` that would be nice to be tested.

Reply:
I agree, however, at the moment, this effort is a POC and requirements are subject to change. I will add more tests once we decide to make it a feature.